app/models/answer.rb

```diff
@@ -3,4 +3,13 @@ class Answer < ApplicationRecord
   belongs_to :user
 
   validates :body, :question_id, presence: true
+  validates :best, uniqueness: { scope: :question_id }, if: :best
+
+  def best!
+    old_best = question.answers.where(best: true)
+    transaction do
+      old_best.update(best: false)
```

Comment:
:(     
NoMethodError:
       undefined method `update!' for #<ActiveRecord::AssociationRelation []>
       Did you mean?  update